### PR TITLE
Add accessibility tests for experimental

### PR DESCRIPTION
**Pull request by sophschneider**
### WHY are these changes introduced?

Fixes [this issue
](https://github.com/Shopify/polaris-summer-editions/issues/1030)

### WHAT is this pull request doing?

Adds a duplicate CI check to run the accessibility tests on storybook with the `polarisSummerEditions2023` feature flag forced to true. I do this with a [Storybook Env Variable](https://storybook.js.org/docs/react/configure/environment-variables).

I also had to fix some minor a11y failures, which included adding an `aria-role` to `VerticalStack`

Screenshot 2023-07-28 at 10 23 02 AM

### How to 🎩

* Make sure the new CI passes!
* `aria-role='menu'` is on `Box` pre se23 and on `VerticalStack` post se23 in [`ActionList`](https://5d559397bae39100201eedc1-fbggpulucd.chromatic.com/?path=/story/all-components-actionlist--all)

### CODE REVIEW

**Review by sam-b-rose — APPROVED** [review]
Looks great! Thank you for fixing this 👏 💯

**Comment by sam-b-rose** [comment]
We will eventually want to remove this additional CI check once we are ready to make se23 the default. I think it is a good idea to keep testing both for a bit.

**Review by laurkim — APPROVED** [review]
💯 🙌

File: .github/workflows/ci-a11y-vrt-experimental.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
+name: Accessibility and Visual Regression Experimental Tests
+
+on:
+  push:
+    branches:
+      - main
+      - next
```
Comment (sam-b-rose) [inline .github/workflows/ci-a11y-vrt-experimental.yml:7]: We've used `next` as the branch name for the next major release. However, I recently created a `beta` branch for automatically publishing prereleases with the `beta` tag.

Should we start using `beta` here or should we make the `next` brach publish beta prereleases?

Comment (sophschneider) [reply .github/workflows/ci-a11y-vrt-experimental.yml:7]: I can update this to `beta`, thanks for the context!

Comment (sophschneider) [reply .github/workflows/ci-a11y-vrt-experimental.yml:7]: I can update the other one as well

Comment (sophschneider) [reply .github/workflows/ci-a11y-vrt-experimental.yml:7]: I kept both :)

**Review by aaronccasanova — APPROVED** [review]

**Pull request merged**